model/src/main/proto/topic.proto

```diff
@@ -238,9 +246,33 @@ message TopicProgress {
   string topic_id = 1;
 
   // Map from story ID to StoryProgress.
-  map<string, StoryProgress> story_progress = 2;
+  map<string, StoryProgress> story_progress = 3;
+
+  // Indicates type of story promoted
+   PromotedStoriesType promoted_stories_type = 4;
+
+  // Timestamp to record last time the exploration was played in ms.
+  int64 last_played_timestamp = 5;
 }
 
+// A structure corresponding to the Promoted Stories. This structure is set up
+// to properly account for recently_played stories, for recommended stories
+// and for coming soon topics.
+message PromotedStoriesType {
```

Comment:
As discussed during the meeting, there are a few things we change here for correctness & simplicity. Without the documentation, suggest:

```proto
message RecommendedActivityList {
  oneof recommendation_type {
    RecommendedStoryList recommended_story_list = 1;
    ComingSoonTopicList coming_soon_topic_list = 2;
  }
}

message RecommendedStoryList {
  repeated PromotedStory recently_played = 1;
  repeated PromotedStory suggest_story = 2;
}

message ComingSoonTopicList {
  repeated UpcomingTopic upcoming_topic = 1;
}

...

message UpcomingTopic {
  string topic_id = 1;
  string name = 2;
  int32 version = 3;
  int64 estimated_release_unix_timestamp = 4;
  LessonThumbnail lesson_thumbnail = 5;
}
```

Subsequently, other protos will need to change here.

Reply:
Suggest updating the RecommendedStoryList to 
```
message RecommendedStoryList {
// Ongoing stories from within the last 7 days.
  repeated PromotedStory recent_story_played = 1;

// Other ongoing stories from longer than 7 days ago.
  repeated PromotedStory old_story_played = 2;

  repeated PromotedStory suggest_story = 3;
}
```

or 

```
message RecommendedStoryList {
// Ongoing stories from within the last 7 days.
  repeated PromotedStory recently_played_story = 1;

// Other ongoing stories from longer than 7 days ago.
  repeated PromotedStory older_played_story = 2;

  repeated PromotedStory suggest_story = 3;
}
```

Reply:
Done.

Reply:
Done. Resolving comment.